docs/parse_argparse.py

```diff
@@ -0,0 +1,315 @@
+                # remember path_main_file is remaining constant for one dir_path
+                if path_recorded == path_main_file:
+                    continue
+                # this counts the number of sub-directories each directory has
+                # true only for `common` (as of April 24, 2020)
+                if len(dirnames) > 0:
+                    for dirname in dirnames:
+                        if "__" in dirname:  # to prevent __pychache__ folders from being considered
+                            continue
+                        dir_main = dirname + "_main"  # external_main
+                        path_recorded = path_main_file
+                        lines = f"\n\t {dirname}/{dir_main}"
+                        outlines += lines
+            fh.write(outlines)
+    # Opens each DirectoryName_main.rst and add the name of the scripts in that directory
+    with open(path_main_file, "a") as fh:
+        fname = os.path.basename(fpath)
+        # extract the "kmers" of kmers.py
+        basename = os.path.splitext(fname)[0]
+        fh.write(f"\n\t {basename}")
+
+
+def link_write_usage():
+    """
+    Creates usage.rst and writes the directories in its toctree
+    """
+    if not os.path.exists(path_usage):
+        with open(path_usage, "w") as fh:
+            write_text = "\n".join(["="*len("usage"), "Usage", "="*len("usage"), " ", ".. toctree::",
+                                    "\t :maxdepth: 3", "\t :caption: Table of Contents", " "])
+            fh.write(write_text)
+            # This can also be done using next() -> https://stackoverflow.com/a/142535/12671809 ,
+            # but reduces readability. What to do ??
+            for dirname in os.listdir(path_scripts):
+                path_dirname = os.path.join(path_scripts, dirname)
+                if os.path.isdir(path_dirname):
+                    dirname_rst = dirname + "_main"
+                    fh.write(f"\n\t {dirname}/{dirname_rst}")
+
+
+def remove_existing_docs():
+    """
+    Removes the exists documentatio in `scripts` directory
+    """
+    if os.path.exists(path_scripts):
+        shutil.rmtree(path_scripts)
+
+
+def remove_empty_dir():
+    """
+    Removes any empty directory that may have been added by mistake
+    """
+    for root, dirnames, filenames in os.walk(os.path.join(os.path.dirname(os.path.dirname(__file__)), "autometa")):
+        if os.listdir(root) == []:
+            os.rmdir(root)
+
+
+def remove_temp_files(tmp_fpath):
+    """
+    Removes the temporary files to which the argparse block was copied
+
+    Parameters
+    ----------
+    tmp_fpath : string
+        path to temporary file that has the argparse block
+        e.g. '</path/to/tmp/file>'
+    """
+    os.remove(tmp_fpath)
+
+
+path_scripts = os.path.join(os.path.dirname(__file__), "source", "scripts")
+path_usage = os.path.join(path_scripts, "usage.rst")
```

Comment:
`path_X` is ambiguous, update with either `X_dirpath`, `X_dpath`, or `X_fpath` accordingly. To help when thinking about variable naming, try to read the code off in your head as a readable sentence. If it sounds like you are just reading lines from a book, you are on the right track.

`path_usage` reads:
"I'm retrieving the path usage" 

`usage_fpath` reads:
"I'm retrieving the usage file path"